plugins/modules/keycloak_userprofile.py

```diff
@@ -0,0 +1,608 @@
+    config:
+      kc.user.profile.config:
+        - unmanagedAttributePolicy: ENABLED
+
+# By setting `Only administrators can write`, unmanaged attributes can be managed only through the administration console and API, useful if you have already
+# defined any custom attribute that can be managed by users but you are unsure about adding other attributes that should only be managed by administrators.
+- name: Enable ADMIN_EDIT on Unmanaged Attributes
+  community.general.keycloak_userprofile:
+    state: present
+    parent_id: master
+    provider_id: declarative-user-profile
+    config:
+      kc.user.profile.config:
+        - unmanagedAttributePolicy: ADMIN_EDIT
+
+# By setting `Only administrators can view`, unmanaged attributes are read-only and only available through the administration console and API.
+- name: Enable ADMIN_VIEW on Unmanaged Attributes
+  community.general.keycloak_userprofile:
+    state: present
+    parent_id: master
+    provider_id: declarative-user-profile
+    config:
+      kc.user.profile.config:
+        - unmanagedAttributePolicy: ADMIN_VIEW
+'''
+
+RETURN = '''
+msg:
+  description: The output message generated by the module.
+  returned: always
+  type: str
+  sample: UserProfileProvider created successfully
+data:
+  description: The data returned by the Keycloak API.
+  returned: when state is present
+  type: dict
+  sample: {...}
+'''
+
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from copy import deepcopy
+import json
+
+
+def main():
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        state=dict(type='str', choices=['present', 'absent'], default='present'),
+        parent_id=dict(type='str', required=True),
+        provider_id=dict(type='str', default='declarative-user-profile', choices=['declarative-user-profile']),
+        config=dict(
+            type='dict',
+            required=False,
+            options={
+                "kc.user.profile.config": dict(
+                    type='list',
+                    elements='dict',
+                    options={
+                        'attributes': dict(
+                            type='list',
+                            elements='dict',
+                            required=False,
+                            options={
+                                'name': dict(type='str', required=True),
+                                'displayName': dict(type='str', required=True),
+                                'validations': dict(
+                                    type='dict',
+                                    # TODO: template this too but without passing null on unrequired empty key/value pairs
+                                    # options={
```

Comment:
Basically if `options` isn't added now, it can never be added in the future.

What exactly is the problem here? That Ansible automatically adds `None` values for all fields the user did not provide? If that's the problem, you have to filter out these values yourself.